benchmark/metrics/test/test_cputime.py

```diff
@@ -0,0 +1,219 @@
+        Line that doesn't fit expected pattern raises exception.
+        """
+        parser = CPUParser()
+        with self.assertRaises(ValueError) as e:
+            parser.lineReceived('Unexpected Error Message')
+        self.assertEqual(e.exception.args[-1], 'Unexpected Error Message')
+
+    def test_unexpected_parse(self):
+        """
+        Line that has incorrectly formatted time raises exception.
+        """
+        parser = CPUParser()
+        with self.assertRaises(ValueError) as e:
+            parser.lineReceived('proc 20:34')
+        self.assertEqual(e.exception.args[-1], 'proc 20:34')
+
+
+class _LocalRunner:
+    """
+    Like SSHRunner, but runs command locally.
+    """
+
+    def _run(self, command_args, handle_stdout):
+        output = subprocess.check_output(command_args)
+        for line in output.split('\n'):
+            handle_stdout(line)
+
+    def run(self, node, command_args, handle_stdout):
+        # Ignore the `node` parameter, we just run locally.
+        return deferToThread(self._run, command_args, handle_stdout)
+
+
+class GetNodeCPUTimeTests(TestCase):
+    """
+    Test ``get_node_cpu_times`` command.
+    """
+
+    def test_get_node_cpu_times(self):
+        """
+        Success results in output of dictionary containing process names.
+        """
+        d = get_node_cpu_times(
+            _LocalRunner(),
+            Node(uuid=uuid4(), public_address=IPAddress('10.0.0.1')),
+            ['init'],
+        )
+
+        def check(result):
+            self.assertEqual(result.keys(), ['init'])
+
+        d.addCallback(check)
+
+        return d
+
+    def test_no_such_process(self):
+        """
+        Errors result in output of ``None``.
+        """
+        d = get_node_cpu_times(
+            _LocalRunner(),
+            Node(uuid=uuid4(), public_address=IPAddress('10.0.0.1')),
+            ['n0n-exist'],
+        )
+
+        def check(result):
+            self.assertIs(result, None)
+
+        d.addCallback(check)
+
+        return d
+
+
+class ComputeChangesTests(SynchronousTestCase):
+    """
+    Test computation of CPU time change between two measurements.
+    """
+
+    def test_compute_change(self):
+        """
+        Process measurements are handled correctly.
+        """
+        labels = ['node1', 'node2']
+        before = [{'foo': 3, 'bar': 5}, {'foo': 10, 'bar': 2}]
+        after = [{'foo': 4, 'bar': 5}, {'foo': 12, 'bar': 5}]
+        result = compute_change(labels, before, after)
+        self.assertEqual(result, {
+            'node1': {'foo': 1, 'bar': 0},
+            'node2': {'foo': 2, 'bar': 3},
+            })
+
+    def test_compute_change_new_proc(self):
+        """
+        Process that only appears in ``after`` is ignored.
+        """
+        labels = ['node1']
+        before = [{'foo': 500}]
+        after = [{'foo': 555, 'bar': 5}]
+        result = compute_change(labels, before, after)
+        self.assertEqual(result, {'node1': {'foo': 55}})
```

Comment:
Given that we are testing that processes that appear in the after list are ignored, would it be better to test for the absence of `bar` rather than also checking that the value for `foo` has been calculated correctly?